packages/app-builder-lib/src/targets/AppxTarget.ts

```diff
@@ -199,19 +224,57 @@ export default class AppXTarget extends Target {
-          const result = options.applicationId || options.identityName || appInfo.name
-          if (!isNaN(parseInt(result[0], 10))) {
-            let message = `AppX Application.Id can’t start with numbers: "${result}"`
-            if (options.applicationId == null) {
-              message += `\nPlease set appx.applicationId (or correct appx.identityName or name)`
+          const validCharactersRegex = /^([A-Za-z][A-Za-z0-9]*)(\.[A-Za-z][A-Za-z0-9]*)*$/
+          const identitynumber = parseInt(options.identityName as string, 10) || NaN
+		  let result:string
+          if (!isNaN(identitynumber) && options.identityName !== null && options.identityName !== undefined) {
+            if (options.identityName[0] === "0") {
+              log.warn(`Remove the 0${identitynumber}`)
+              result = options.identityName.replace("0"+identitynumber.toString(), "")
+            } else {
+              log.warn(`Remove the ${identitynumber}`)
+              result = options.identityName.replace(identitynumber.toString(), "")
             }
+          } else {
+            result = options.applicationId || options.identityName || appInfo.name
+          }
+
+          if (result.length < 1 || result.length > 64) {
+            const message = `Appx Application.Id with a value between 1 and 64 characters in length`
+            throw new InvalidConfigurationError(message)
+          } else if (!validCharactersRegex.test(result)) {
+            const message = `AppX Application.Id can not be consists of alpha-numeric and period"`
+            throw new InvalidConfigurationError(message)
+          } else if (restrictedApplicationIdValues.includes(result.toUpperCase())) {
+            const message = `AppX Application.Id can not be some values`
+            throw new InvalidConfigurationError(message)
+          } else if (result == null && options.applicationId == null) {
+            const message = `Please set appx.applicationId (or correct appx.identityName or name)`
             throw new InvalidConfigurationError(message)
           }
+
           return result
         }
 
-        case "identityName":
-          return options.identityName || appInfo.name
+        case "identityName": {
+          const result = options.identityName || appInfo.name
+          const validCharactersRegex = /^[a-zA-Z0-9.-]+$/
+          if (result.length < 3 || result.length > 50) {
+            const message = `Appx identityName.Id with a value between 3 and 50 characters in length`
+            throw new InvalidConfigurationError(message)
+          } else if (!validCharactersRegex.test(result)) {
+            const message = `AppX identityName.Id cat be consists of alpha-numeric, period, and dash characters"`
+            throw new InvalidConfigurationError(message)
+          } else if (restrictedApplicationIdValues.includes(result.toUpperCase())) {
+            const message = `AppX identityName.Id can not be some values`
```

Comment:
Let's be more explicit in our error message in this case otherwise users won't know what was wrong with their Identity Name

Maybe we can use?
```
const message = `AppX identityName.Id can not include ${JSON.stringify(restrictedApplicationIdValues)}`
```